app/controllers/suggestions_controller.rb

```diff
@@ -0,0 +1,50 @@
+require 'yelp'
+
+class SuggestionsController < ApplicationController
+
+  def show
+    params[:limit] = 5 if params[:limit] == ""
+    params[:limit] = 20 if params[:limit].to_i > 20
+    @search = TunesTakeoutWrapper.search(params[:term], params[:limit])
+    unless @search.nil?
+      @suggestions = @search["suggestions"]
+      @pairings = turn_suggestions_into_instances(@suggestions)
+    end
+    @favorites = TunesTakeoutWrapper.favorite_ids(current_user.uid) if current_user
```

Comment:
What happens if current_user is nil? Does this work? Sorry I can't get the app to run locally because I don't have API keys so I'm not sure. Just something to be mindful of when using if statements in this format (will this throw an error if my variable doesn't exist?)